Library/Homebrew/formula.rb

```diff
@@ -2766,8 +2766,15 @@ def pour_bottle?(&block)
     # <pre>deprecate! date: "2020-08-27", because: "has been replaced by foo"</pre>
     def deprecate!(date: nil, because: nil)
       odeprecated "`deprecate!` without a reason", "`deprecate! because: \"reason\"`" if because.blank?
+      odeprecated "`deprecate!` without a date", "`deprecate! date: ...`" if date.blank?
 
-      return if date.present? && Date.parse(date) > Date.today
+      if date.present? && (parsed_date = Date.parse(date)) && parsed_date > Date.today
+        if date != parsed_date.to_s
+          odeprecated "`disable!` without an ISO 8601 date", "`disable! date: \"#{parsed_date}\"`"
```

Comment:
```suggestion
          odeprecated "`deprecate!` without an ISO 8601 date", "`deprecate! date: \"#{parsed_date}\"`"
```

Reply:
Also, I'd think that we'd want to check the date format even if we haven't yet reached the date, right?

Reply:
I'm not sure this deprecation is even needed. While we prefer ISO 8601, the functionality will not be affected by a non-ISO 8601 date. Additionally, once it's been deprecated and disabled, it will just end up being re-enabled. So I think a style check (which we already have) is a better choice here.

Reply:
> I'm not sure this deprecation is even needed.
> So I think a style check (which we already have) is a better choice here.

Agreed 👍🏻